src/pages/public_registration/action_accordions/DeletePublication.tsx

```diff
@@ -0,0 +1,148 @@
+import { Registration, UnpublishPublicationRequest } from '../../../types/registration.types';
+import { Box, Breadcrumbs, Button, DialogActions, Divider, TextField, Typography } from '@mui/material';
+import { useState } from 'react';
+import { Modal } from '../../../components/Modal';
+import { ErrorMessage, Field, FieldProps, Form, Formik } from 'formik';
+import * as Yup from 'yup';
+import { useTranslation } from 'react-i18next';
+import { FindRegistration } from './FindRegistration';
+import { unpublishRegistration } from '../../../api/registrationApi';
+import { setNotification } from '../../../redux/notificationSlice';
+import { useDispatch } from 'react-redux';
+import { LoadingButton } from '@mui/lab';
+import { RequiredDescription } from '../../../components/RequiredDescription';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { useHistory } from 'react-router-dom';
+import i18n from '../../../translations/i18n';
+import { useMutation } from '@tanstack/react-query';
+
+interface DeleteForm {
+  deleteMessage: string;
+}
+
+interface DeletePublicationProps {
+  registration: Registration;
+}
+
+export const DeletePublication = ({ registration }: DeletePublicationProps) => {
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const [selectedDuplicate, setSelectedDuplicate] = useState<Registration | null>(null);
+  const history = useHistory();
+
+  const deleteValidationSchema = Yup.object().shape({
+    deleteMessage: Yup.string()
+      .min(3, t('feedback.validation.must_be_bigger_than', { field: i18n.t('common.justification'), limit: 3 }))
+      .required(t('feedback.validation.is_required', { field: i18n.t('common.justification') })),
+  });
+
+  const unpublishRegistrationMutation = useMutation({
+    mutationFn: (unpublishRequest: UnpublishPublicationRequest) =>
+      unpublishRegistration(registration.identifier, unpublishRequest),
```

Comment:
kan fjerne behovet for dobbelt-wrapped funksjonskallet med `handleDelete` med noe sånt:

```suggestion
    mutationFn: (values: DeleteForm) => {
      if (selectedDuplicate) {
        return unpublishRegistration(registration.identifier, {
          type: 'UnpublishPublicationRequest',
          duplicateOf: selectedDuplicate.id,
          comment: values.deleteMessage,
        });
      } else {
        return unpublishRegistration(registration.identifier, {
          type: 'UnpublishPublicationRequest',
          comment: values.deleteMessage,
        });
      }
    },
```
Og endre onSubmit til: `onSubmit={(values) => unpublishRegistrationMutation.mutate(values)}`

Færre funksjoner, mindre kode, og IMO litt mer lesbart :)